### DISCUSSION TITLE
docs/query-conditions

### DISCUSSION THREAD

**Discussion by giscus** [mikro-orm/mikro-orm#5194] [Comments]
# docs/query-conditions

When you want to make complex queries, we can easily end up with a lot of boilerplate code full of curly brackets:

https://mikro-orm.io/docs/query-conditions

**Comment by dyedwiper** [comment]
I recognized that the `$in` operator also works with an array as input, checking if the intersection of both arrays is not empty.
```
const entities = em.find(SomeEntity, { someArrayProp: { $in: someArray } });
// returns entities where intersection of someArrayProp and someArray is not empty
```
This is quite convenient for me. Maybe you want to add this to the documentation...
(Tried this with MongoDriver btw.)

**Reply by B4nan** [reply]
That sounds like a mongo specific behavior, otherwise I know this is supported in postgres through one of its special array operator (e.g. `$contains)

**Comment by pasivad** [comment]
Is there a way to check if Collection contains specific element or not, using older MicroOrm versions(without collection operators)? 
If collection contains only 1 entity we can do it with $eq operator, but if collection will consists of 2 or more entities, $eq operator will work in an inappropriate way. It will always return false because we have many elements 
Example: `{ shipmentRequests: { shipment: { $ne: shipment } } }`
ShipmentRequests - collection where we have ShipmentEM, so we need to check if at least one shipment is not equal
